# Fresh Market EDA:

In [2]:
import pandas as pd
import seaborn as sns



In [3]:
data = pd.read_pickle(r"C:\Users\Jahir\DataspellProjects\FreshMarket-project\datos\fresh_market-limpio.pkl")
fresh = data.copy()

In [4]:
fresh.info()

<class 'pandas.core.frame.DataFrame'>
Index: 999 entries, 0 to 999
Data columns (total 17 columns):
 #   Column                   Non-Null Count  Dtype         
---  ------                   --------------  -----         
 0   invoice_id               999 non-null    string        
 1   branch                   999 non-null    category      
 2   city                     999 non-null    string        
 3   customer_type            999 non-null    string        
 4   gender                   999 non-null    category      
 5   product_line             999 non-null    string        
 6   unit_price               999 non-null    float64       
 7   quantity                 999 non-null    int64         
 8   tax_5%                   999 non-null    float64       
 9   total                    999 non-null    float64       
 10  date                     999 non-null    datetime64[ns]
 11  time                     999 non-null    object        
 12  payment                  999 non-null    

Nuestro dataset consta de 17 columnas y un total de 999 filas sin valores faltantes, el formato de cada columna se verifica que esta correcto. En la siguiente linea hacemos una descripcion estadistica general de todas nuestras variables numericas.

In [9]:
fresh[['unit_price', 'quantity', 'tax_5%', 'total','payment', 'cogs', 'gross_margin_percentage', 'gross_income','rating']].describe()

unit_price    quantity      tax_5%        total        cogs  \
count  999.000000  999.000000  999.000000   999.000000  999.000000   
mean    55.296486    5.505506   15.381061   322.999389  307.618398   
std     26.878903    2.921436   11.714604   246.006398  234.291748   
min      0.170000    1.000000    0.510000    10.680000   10.170000   
25%     32.000000    3.000000    5.895000   123.790000  117.895000   
50%     55.000000    5.000000   12.080000   253.680000  241.600000   
75%     78.000000    8.000000   22.460000   471.670000  449.210000   
max    100.000000   10.000000   49.650000  1042.650000  993.000000   

       gross_margin_percentage  gross_income      rating  
count                   999.00    999.000000  999.000000  
mean                      4.76     15.381061    6.970771  
std                       0.00     11.714604    1.718357  
min                       4.76      0.510000    4.000000  
25%                       4.76      5.895000    5.500000  
50%                       4.76     12.080000    7.000000  
75%                       4.76     22.460000    8.500000  
max                       4.76     49.650000   10.000000

Valores estan dentro del rango normal establecido,sin outliers destacables. Nuestros datos contienens una divicion de poblacion basica por sexo, vamos a crear una tabla de frecuencia y ver la distribucion.

In [11]:
fresh

invoice_id branch       city customer_type  gender  \
0    750-67-8428      A     Yangon        Member  Female   
1    226-31-3081      C  Naypyitaw        Normal  Female   
2    631-41-3108      A     Yangon        Normal    Male   
3    123-19-1176      A     Yangon        Member    Male   
4    373-73-7910      A     Yangon        Normal    Male   
..           ...    ...        ...           ...     ...   
995  233-67-5758      C  Naypyitaw        Normal    Male   
996  303-96-2227      B   Mandalay        Normal  Female   
997  727-02-1313      A     Yangon        Member    Male   
998  347-56-2442      A     Yangon        Normal    Male   
999  849-09-3807      A     Yangon        Member  Female   

               product_line  unit_price  quantity  tax_5%    total       date  \
0         Health and beauty        75.0         7   26.14   548.97 2019-01-05   
1    Electronic accessories        15.0         5    3.82    80.22 2019-03-08   
2        Home and lifestyle        46.0         7   16.22   340.53 2019-03-03   
3         Health and beauty        58.0         8   23.29   489.05 2019-01-27   
4         Sports and travel        86.0         7   30.21   634.38 2019-02-08   
..                      ...         ...       ...     ...      ...        ...   
995       Health and beauty        40.0         1    2.02    42.37 2019-01-29   
996      Home and lifestyle        97.0        10   48.69  1022.49 2019-03-02   
997      Food and beverages        32.0         1    1.59    33.43 2019-02-09   
998      Home and lifestyle        66.0         1    3.29    69.11 2019-02-22   
999     Fashion accessories        88.0         7   30.92   649.30 2019-02-18   

            time      payment    cogs  gross_margin_percentage  gross_income  \
0     1:08:00 PM      Ewallet  522.83                     4.76         26.14   
1    10:29:00 AM         Cash   76.40                     4.76          3.82   
2     1:23:00 PM  Credit card  324.31                     4.76         16.22   
3     8:33:00 PM      Ewallet  465.76                     4.76         23.29   
4    10:37:00 AM      Ewallet  604.17                     4.76         30.21   
..           ...          ...     ...                      ...           ...   
995   1:46:00 PM      Ewallet   40.35                     4.76          2.02   
996   5:16:00 PM      Ewallet  973.80                     4.76         48.69   
997   1:22:00 PM         Cash   31.84                     4.76          1.59   
998   3:33:00 PM         Cash   65.82                     4.76          3.29   
999   1:28:00 PM         Cash  618.38                     4.76         30.92   

     rating  
0       9.1  
1       9.6  
2       7.4  
3       8.4  
4       5.3  
..      ...  
995     6.2  
996     4.4  
997     7.7  
998     4.1  
999     6.6  

[999 rows x 17 columns]

In [ ]:
sns.histplot(data=fresh, x=fresh['branch'])